src/languages/objectivec.js

```diff
@@ -69,16 +69,24 @@ function(hljs) {
       },
       {
         className: 'meta',
-        begin: '#',
-        end: '$',
+        begin: /#\s*[a-z]+\b/, end: /$/,
+        keywords: {
+          'meta-keyword':
+            'if else elif endif define undef warning error line ' +
+            'pragma ifdef ifndef include'
+        },
         contains: [
+          {
+            begin: /\\\n/, relevance: 0
+          },
+          hljs.inherit(hljs.QUOTE_STRING_MODE, {className: 'meta-string'}),
           {
             className: 'meta-string',
-            variants: [
-              { begin: '\"', end: '\"' },
-              { begin: '<', end: '>' }
-            ]
-          }
+            begin: /<[^\n>]*>/, end: /$/,
```

Comment:
Would a non greedy ‘.*?’ work here instead?

Reply:
Seems to work. This is taken from the cpp definition, so if we change one we should probably change the other one too. Do you have a preference?

Reply:
Yeah we’d change both. I think the non greedy is a nicer and more elegant way to write it.

Reply:
Done. (Probably worth noting that non-greedy `.*?` has more potential to make the regex go nuts with backtracking or otherwise misbehave. But, in this case, since there is only one byte after the pattern, it's fine.)

Reply:
> Probably worth noting that non-greedy .*? has more potential to make the regex go nuts with backtracking or otherwise misbehave

Is there a reference or any article you could point me to on this issue?

Reply:
Not off-hand. Searching around I found this: http://blog.stevenlevithan.com/archives/greedy-lazy-performance

Though I was thinking a slightly different issue. Suppose you wanted to find empty divs, accepting random attributes. `<div .*?></div>` would misbehave a bit. Given `<div id=a>blahblahblah<div id=b></div>`, the regex would match the whole string rather than just the ending bit. Also if there are a lot of strings that almost match `></div>`, the regex might go down that route and backtracking a bunch. (Note sure whether JS implementations tend to use backtracking or DFA-based regex engines.)